openlibrary/core/cache.py

```diff
@@ -177,20 +178,20 @@ def join_threads(self):
             thread.join()
 
     def encode_args(self, args, kw=None):
+        """Encodes arguments to construct the memcache key."""
         kw = kw or {}
-        """Encodes arguments to construct the memcache key.
-        """
+
         # strip [ and ] from key
         a = self.json_encode(list(args))[1:-1]
 
         if kw:
-            return a + "-" + self.json_encode(kw)
+            return f"{hashlib.md5(a.encode('utf-8')).hexdigest()}${hashlib.md5(self.json_encode(kw).encode('utf-8')).hexdigest()}"
```

Comment:
Like you mentioned I think hashing could cause some problems with security/perf, so I think we might want it to optionally hash as an argument, defaulting to false. That way for cases where a collision doesn't matter, like query carousels, we can hash. Otherwise we can collide and return potentially private data!

Also, in the case where do hash, I reckon we can combine `a` and `kw`, eg `hash_fn(json.dumps([a, kw]))` and hash them together. One less md5 call!

Reply:
agreed that 1 hash call will be more performant at scale

Reply:
@jimchamp @mek just to clarify are we clear on the other proposal I made here? about making the hashing an option?

This method is in openlibrary/core/cach.py, and is used by a lot of different places where we run caching. It could have security implications in certain spots. We likely want to only hash the caching key on a case-by-case basis to be safe!